test/e2e/antreapolicy_test.go

```diff
@@ -1039,30 +1093,65 @@ func testACNPClusterGroupRefRulePodAdd(t *testing.T, data *TestData) {
 	executeTestsWithData(t, testCase, data)
 }
 
+// There are three situations of a Pod's IP(s):
+// 1. Only one IPv4 address.
+// 2. Only one IPv6 address.
+// 3. One IPv4 and one IPv6 address, and we don't know the order in list.
+// We need to add all IP(s) of Pods as CIDR to IPBlock.
+func genIPBlock(ip string) *crdv1beta1.IPBlock {
+	switch IPFamily(ip) {
+	case "v4":
+		return &crdv1beta1.IPBlock{CIDR: ip + "/32"}
+	case "v6":
+		return &crdv1beta1.IPBlock{CIDR: ip + "/128"}
+	default:
+		return nil
+	}
+}
+
+// genIPBlockWithExceptIPs generates ipBlocks which contains all the IP addresses in the
+// provided IPs' address family(s), except for the addresses in the input slice.
+func genIPBlockWithExceptIPs(except []string) []*crdv1beta1.IPBlock {
+	var v4Excepts, v6Excepts []string
+	var ipbs []*crdv1beta1.IPBlock
+	for _, e := range except {
+		if IPFamily(e) == "v4" {
+			v4Excepts = append(v4Excepts, e+"/32")
+		} else if IPFamily(e) == "v6" {
+			v6Excepts = append(v6Excepts, e+"/128")
+		}
+	}
+	if len(v4Excepts) > 0 {
+		ipbs = append(ipbs, &crdv1beta1.IPBlock{
+			CIDR:   anyIPv4,
```

Comment:
would it make more sense to use the podCIDR for the cluster here instead of `anyIPv4`? Or do we have some external traffic as well?

Reply:
I don't think we have any testcase using podCIDR as of now, which is very tricky to obtain for e2e tests running on public cloud

Reply:
Our e2e tests already assume that this is available. See `podV4NetworkCIDR` / `podV6NetworkCIDR`.

Reply:
What I meant is that those CIDRs are only used as of now to determine whether a cluster is ipv4 or ipv6 addressed (or both), so as to determine what specific test steps to run. But they are not used in policy rules at all. We could add checks like `if clusterInfo.podV4NetworkCIDR != ""` like current occurences, but I'm thinking that in public cloud e2e tests if the framework was not able to retrieve cluster cidr info then the test will not be run there

Reply:
I don't think we run our e2e tests in public cloud and IIRC our e2e tests will refuse to run if the pod CIDR(s) cannot be retrieved.
Anyway, purely from a testing perspective, I imagine there is no advantage to using the pod CIDR as opposed to `0/0`.